### ISSUE TITLE
Model updates are not being tracked

### ISSUE THREAD

**Issue by hslzr** [mongoid/mongoid-history#184]
I followed the basic guide to get started, but I noticed the `update` actions are not being recorded. These are my models:

```ruby
class Activist
  include Mongoid::Document
  include Mongoid::Timestamps
  include Mongoid::History::Trackable

  track_history on: :fields,
      modifier_field: :modifier,
      version_field: :version,
      track_create: true,
      track_update: true,
      track_destroy: true
end

class Citizen
  include Mongoid::Document
  include Mongoid::Timestamps
  include Mongoid::History::Trackable

  track_history on: :fields,
      scope: :activist,
      version_field: :version,
      track_create: true,
      track_update: true,
      track_destroy: true
end
```

Now, in the rails console:

```
activist = Activist.create ( attributes )

citizen = activist.citizens.create ( attributes )

citizen.history_tracks.count
# => 1

citizen.update_attributes(name: "something")
#=> true

citizen.history_tracks.count
# => 1     should be 2

citizen.history_tracks.last.action
#=> "create"
# Shouldn't it be "update" ?

citizen.destroy
# => true

citizen.history_tracks.count
# => 2

citizen.history_tracks.last.action
# => "destroy"
```

Am I missing something? I also created the initializer as provided by the readme.

**Comment by dblock** [comment]
From the first view this should work. Activist is not persisted, I wonder if it has something to do with it? Use `!` functions to see if there's another in the middle. And change to `Activist.create!(attributes)` and `update_attributes!`. Finally, what's the relation between citizen and activist?

If you can't figure it out, turn it into a spec in this repo.

**Comment by hslzr** [comment]
I ran it in the terminal and yes, the error was `Activist` model not being persisted, but still, when I try with this spec this is not working:

```ruby
describe '#history' do
  let(:user) { create(:user) }
  let(:activist) { create(:activist, modifier: user }
  let(:citizen) { create(:citizen, activist: activist, modifier: user) }

  it 'tracks citizen updates' do
    citizen.update_attributes(first_name: "John")
    expect(citizen.history_tracks.last.action).to eq("update")
  end
end
```

The error is I'm still receiving `"create"` in this spec.

**Comment by dblock** [comment]
First, again, use `!` versions of these methods, you might be not seeing an exception.

`.last` in Mongoid is not a sorted query, so I wouldn't be so sure that `.last` is the last thing, see if they are in there out of order and whether `citizen.history_tracks.asc(:_id).last` is the right thing?

**Comment by hslzr** [comment]
My problem was somewhere else, I guess.

I looked into the specs and found the  `tracked_fields` method. I ran it into my `Activist` model, only to find out it was returning an empty array, and so was doing `Citizen`.

The solution was to move the `track_on` method _**below**_ the declaration of the fields. As of why it worked on the rails console and not in the specs, beats me; but changing that made my specs go green again.

Thank you for your support!

**Comment by dblock** [comment]
Interesting. We definitely should be failing when you try to `track_fields` on a field that doesn't exist! Or it should work. 

Would you mind trying to write a spec in this repo for that scenario and making a PR with it? Maybe it's a real bug.

**Comment by hslzr** [comment]
Sure, I'll give it a look this afternoon.

**Comment by mpetazzoni** [comment]
Looks to be a dup of #174 then?

**Comment by dblock** [comment]
Oh thanks @mpetazzoni, @hslzr do you define fields after `track_history`?

**Comment by hslzr** [comment]
Yes @mpetazzoni, looks like my question is a duplicate of #174. And that's right @dblock, I was indeed defining the fields _after_ the track history :sweat_smile:

**Comment by dblock** [comment]
Lets close this, maybe @hslzr can feel motivated to make a fix for #174?
